### ISSUE TITLE
[Dataset] read_json doesn't work with zipped json files

### ISSUE THREAD

**Issue by wuisawesome** [ray-project/ray#17296]
### What is the problem?

`read_json('s3://bucket/with/zipped.json.gz')` doesn't work. This is surprising as arrow and pandas support this. 

*Ray version and other system information (Python version, TensorFlow version, OS):*

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by wuisawesome** [comment]
cc @clarkzinzow @ericl

**Comment by clarkzinzow** [comment]
@wuisawesome do you have a traceback or something?

**Comment by clarkzinzow** [comment]
Also I'm pretty sure that reading compressed JSON/CSV files from non-local storage isn't supported by Arrow, or at least isn't supported when giving a `NativeFile` handle instead of a path (which we need to do to support custom filesystems).

**Comment by wuisawesome** [comment]
Here's a simple repro: 

```
df1 = pd.DataFrame({"one": [1, 2, 3], "two": ["a", "b", "c"]})
df1.to_json("data/data0.json.gz", compression="gzip")

t = pyarrow.json.read_json("data/data0.json.gz")
print(t)


# Breaks here.
ds = ray.data.read_json("data/data0.json.gz")
ds.show()
```

**Comment by wuisawesome** [comment]
Stack trace: 
```
Traceback (most recent call last):
  File "temp.py", line 33, in <module>
    ds.show()
  File "/Users/alex/anyscale/ray/python/ray/experimental/data/dataset.py", line 562, in show
    for row in self.take(limit):
  File "/Users/alex/anyscale/ray/python/ray/experimental/data/dataset.py", line 548, in take
    for row in self.iter_rows():
  File "/Users/alex/anyscale/ray/python/ray/experimental/data/dataset.py", line 812, in iter_rows
    prefetch_blocks=prefetch_blocks, batch_format="_blocks"):
  File "/Users/alex/anyscale/ray/python/ray/experimental/data/dataset.py", line 878, in iter_batches
    block = ray.get(block_window[0])
  File "/Users/alex/anyscale/ray/python/ray/_private/client_mode_hook.py", line 82, in wrapper
    return func(*args, **kwargs)
  File "/Users/alex/anyscale/ray/python/ray/worker.py", line 1578, in get
    raise value.as_instanceof_cause()
ray.exceptions.RayTaskError(ArrowInvalid): ray::remote_read() (pid=34410, ip=192.168.2.35)
  File "python/ray/_raylet.pyx", line 534, in ray._raylet.execute_task
  File "/Users/alex/anyscale/ray/python/ray/experimental/data/read_api.py", line 149, in remote_read
    return task()
  File "/Users/alex/anyscale/ray/python/ray/experimental/data/datasource/datasource.py", line 106, in __call__
    return self._read_fn()
  File "/Users/alex/anyscale/ray/python/ray/experimental/data/datasource/file_based_datasource.py", line 58, in <lambda>
    read_paths, filesystem),
  File "/Users/alex/anyscale/ray/python/ray/experimental/data/datasource/file_based_datasource.py", line 52, in read_files
    tables.append(read_file(f, **reader_args))
  File "/Users/alex/anyscale/ray/python/ray/experimental/data/datasource/json_datasource.py", line 25, in _read_file
    f, read_options=read_options, **arrow_reader_args)
  File "pyarrow/_json.pyx", line 247, in pyarrow._json.read_json
  File "pyarrow/error.pxi", line 122, in pyarrow.lib.pyarrow_internal_check_status
  File "pyarrow/error.pxi", line 84, in pyarrow.lib.check_status
pyarrow.lib.ArrowInvalid: JSON parse error: Invalid value. in row 0
```

**Comment by wuisawesome** [comment]
Closed by #17355
